pkg/auth.go

```diff
@@ -227,6 +239,40 @@ func (auth *Auth) ValidateContract(request ValidationRequest) (*ValidationResult
 	return nil, fmt.Errorf("format %v: %w", request.ContractFormat, ErrUnknownContractFormat)
 }
 
+// CreateAccessToken extracts the claims out of the request, checks the validity and builds the access token
+func (auth *Auth) CreateAccessToken(request CreateAccessTokenRequest) (*AccessTokenResponse, error) {
```

Comment:
You might want to group the validation in a separate method so it could be used by other API's (also when access token generation is handled by the vendor itself)

Reply:
Lets do that when use-cases emerge. I find it hard to extract it without knowing how it might be used. Changes are, even when extracted in a seperate method, it still needs work.